// WARNING: DO NOT EDIT THIS FILE. THIS FILE IS MANAGED BY SPRING ROO.
// You may push code into the target .java compilation unit if you wish to edit any member(s).

package com.mcel2.domain;

import com.mcel2.domain.PossibleFailures;
import com.mcel2.domain.PossibleFailuresDataOnDemand;
import com.mcel2.domain.PossibleFailuresIntegrationTest;
import java.util.Iterator;
import java.util.List;
import javax.validation.ConstraintViolation;
import javax.validation.ConstraintViolationException;
import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.transaction.annotation.Transactional;

privileged aspect PossibleFailuresIntegrationTest_Roo_IntegrationTest {
    
    declare @type: PossibleFailuresIntegrationTest: @RunWith(SpringJUnit4ClassRunner.class);
    
    declare @type: PossibleFailuresIntegrationTest: @ContextConfiguration(locations = "classpath*:/META-INF/spring/applicationContext*.xml");
    
    declare @type: PossibleFailuresIntegrationTest: @Transactional;
    
    @Autowired
    PossibleFailuresDataOnDemand PossibleFailuresIntegrationTest.dod;
    
    @Test
    public void PossibleFailuresIntegrationTest.testCountPossibleFailureses() {
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", dod.getRandomPossibleFailures());
        long count = PossibleFailures.countPossibleFailureses();
        Assert.assertTrue("Counter for 'PossibleFailures' incorrectly reported there were no entries", count > 0);
    }
    
    @Test
    public void PossibleFailuresIntegrationTest.testFindPossibleFailures() {
        PossibleFailures obj = dod.getRandomPossibleFailures();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", obj);
        Long id = obj.getId();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to provide an identifier", id);
        obj = PossibleFailures.findPossibleFailures(id);
        Assert.assertNotNull("Find method for 'PossibleFailures' illegally returned null for id '" + id + "'", obj);
        Assert.assertEquals("Find method for 'PossibleFailures' returned the incorrect identifier", id, obj.getId());
    }
    
    @Test
    public void PossibleFailuresIntegrationTest.testFindAllPossibleFailureses() {
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", dod.getRandomPossibleFailures());
        long count = PossibleFailures.countPossibleFailureses();
        Assert.assertTrue("Too expensive to perform a find all test for 'PossibleFailures', as there are " + count + " entries; set the findAllMaximum to exceed this value or set findAll=false on the integration test annotation to disable the test", count < 250);
        List<PossibleFailures> result = PossibleFailures.findAllPossibleFailureses();
        Assert.assertNotNull("Find all method for 'PossibleFailures' illegally returned null", result);
        Assert.assertTrue("Find all method for 'PossibleFailures' failed to return any data", result.size() > 0);
    }
    
    @Test
    public void PossibleFailuresIntegrationTest.testFindPossibleFailuresEntries() {
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", dod.getRandomPossibleFailures());
        long count = PossibleFailures.countPossibleFailureses();
        if (count > 20) count = 20;
        int firstResult = 0;
        int maxResults = (int) count;
        List<PossibleFailures> result = PossibleFailures.findPossibleFailuresEntries(firstResult, maxResults);
        Assert.assertNotNull("Find entries method for 'PossibleFailures' illegally returned null", result);
        Assert.assertEquals("Find entries method for 'PossibleFailures' returned an incorrect number of entries", count, result.size());
    }
    
    @Test
    public void PossibleFailuresIntegrationTest.testFlush() {
        PossibleFailures obj = dod.getRandomPossibleFailures();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", obj);
        Long id = obj.getId();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to provide an identifier", id);
        obj = PossibleFailures.findPossibleFailures(id);
        Assert.assertNotNull("Find method for 'PossibleFailures' illegally returned null for id '" + id + "'", obj);
        boolean modified =  dod.modifyPossibleFailures(obj);
        Integer currentVersion = obj.getVersion();
        obj.flush();
        Assert.assertTrue("Version for 'PossibleFailures' failed to increment on flush directive", (currentVersion != null && obj.getVersion() > currentVersion) || !modified);
    }
    
    @Test
    public void PossibleFailuresIntegrationTest.testMergeUpdate() {
        PossibleFailures obj = dod.getRandomPossibleFailures();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", obj);
        Long id = obj.getId();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to provide an identifier", id);
        obj = PossibleFailures.findPossibleFailures(id);
        boolean modified =  dod.modifyPossibleFailures(obj);
        Integer currentVersion = obj.getVersion();
        PossibleFailures merged = obj.merge();
        obj.flush();
        Assert.assertEquals("Identifier of merged object not the same as identifier of original object", merged.getId(), id);
        Assert.assertTrue("Version for 'PossibleFailures' failed to increment on merge and flush directive", (currentVersion != null && obj.getVersion() > currentVersion) || !modified);
    }
    
    @Test
    public void PossibleFailuresIntegrationTest.testPersist() {
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", dod.getRandomPossibleFailures());
        PossibleFailures obj = dod.getNewTransientPossibleFailures(Integer.MAX_VALUE);
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to provide a new transient entity", obj);
        Assert.assertNull("Expected 'PossibleFailures' identifier to be null", obj.getId());
        try {
            obj.persist();
        } catch (final ConstraintViolationException e) {
            final StringBuilder msg = new StringBuilder();
            for (Iterator<ConstraintViolation<?>> iter = e.getConstraintViolations().iterator(); iter.hasNext();) {
                final ConstraintViolation<?> cv = iter.next();
                msg.append("[").append(cv.getRootBean().getClass().getName()).append(".").append(cv.getPropertyPath()).append(": ").append(cv.getMessage()).append(" (invalid value = ").append(cv.getInvalidValue()).append(")").append("]");
            }
            throw new IllegalStateException(msg.toString(), e);
        }
        obj.flush();
        Assert.assertNotNull("Expected 'PossibleFailures' identifier to no longer be null", obj.getId());
    }
    
    @Test
    public void PossibleFailuresIntegrationTest.testRemove() {
        PossibleFailures obj = dod.getRandomPossibleFailures();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to initialize correctly", obj);
        Long id = obj.getId();
        Assert.assertNotNull("Data on demand for 'PossibleFailures' failed to provide an identifier", id);
        obj = PossibleFailures.findPossibleFailures(id);
        obj.remove();
        obj.flush();
        Assert.assertNull("Failed to remove 'PossibleFailures' with identifier '" + id + "'", PossibleFailures.findPossibleFailures(id));
    }
    
}
